add new API torch.amp.is_autocast_available

guangyey opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #124938

# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available` as a public api.

cc @mcarilli @ptrblck @leslie-fang-intel @jgong5

guangyey commented on torch/__init__.py:
```diff
@@ -1477,7 +1477,7 @@ def manager_path():
     return path.encode('utf-8')
 
-from torch.amp import autocast, GradScaler
+from torch.amp import autocast, GradScaler, is_autocast_available
```
Got it.

guangyey commented on torch/amp/autocast_mode.py:
```diff
@@ -6,7 +6,11 @@
+__all__ = ["autocast_decorator", "autocast", "is_autocast_available"]
+
+
+def is_autocast_available(device_type: str):
```
Add a doc.

albanD reviewed (approved):
SGTM once CI is good!

guangyey committed:
add new API torch.amp.is_autocast_available

[ghstack-poisoned]

guangyey committed:
Update on "add new API torch.amp.is_autocast_available"


# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available`.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

guangyey commented:
@pytorchbot merge

guangyey committed:
Update on "add new API torch.amp.is_autocast_available"


# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available`.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

guangyey committed:
Update on "add new API torch.amp.is_autocast_available"


# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available`.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

guangyey committed:
Update on "add new API torch.amp.is_autocast_available"


# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available`.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

guangyey committed:
Update on "add new API torch.amp.is_autocast_available"


# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available`.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]

guangyey committed:
Update on "add new API torch.amp.is_autocast_available"


# Motivation
expose `torch._is_autocast_available` to `torch.amp.is_autocast_available`.

cc mcarilli ptrblck leslie-fang-intel jgong5

[ghstack-poisoned]
